### PR TITLE
Nathan seeUsersInDashboard Permission

### PR DESCRIPTION
**Pull request by nathanah**
# Description
Adds a permission `seeUsersInDashboard` to replace the role-based logic for allowing admins, owners, and core team members to see all users in the leaderboard and task list on the dashboard.

## Related PRS (if any):
This backend PR is related to the [#2216](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2116) frontend PR.

## Main changes explained:
- Replaced role-based logic with permission-based logic
- Added new permission to be defaultly included in the relevant roles

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as admin user
5. go to dashboard→see all users in leaderboard and task list

### CODE REVIEW

**Review by hbodgal — APPROVED** [review]
Everything looks fine. I tried to change permission to view all users for Volunteer role and was able to see and hide the Team Members by "See All Users in Dashboard and Leaderboard" permission button. I even checked the code. 

image

**Review by ImzIssa — APPROVED** [review]
Hey, I've tested this PR and everything works as expected.
Here is a reference to my frontend PR Review https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2116#pullrequestreview-1983157945

**Review by aaryaneil — APPROVED** [review]
Reviewed your PR and the changes are working as expected. Good work.

Screenshot 2024-04-05 at 12 24 32 PM

**Review by SushmithaPrathap — APPROVED** [review]
Works as expected. 
Here is my frontend PR review: [frontend review](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2116#pullrequestreview-1984418420)

**Review by linkhanthtel — APPROVED** [review]
Looks good! I reviewed your PR and it is working as expected!

**Review by Parth-tech — APPROVED** [review]
Hi Nathan,
Well done. It works perfectly.

[Approved Review](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2116#pullrequestreview-1984467683)

**Review by p-suram — APPROVED** [review]
Changes are working as expected. I have gone through code and tested functionality for Manager and Volunteer roles. I am able to toggle "See All Users in Dashboard and Leaderboard" option and see all users list accordingly.
[Frontend review](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2116#pullrequestreview-1984592012)
image
image
image

File: src/helpers/dashboardhelper.js [hunk 1]
```diff
@@ -217,6 +215,7 @@ const dashboardhelper = function () {
           },
         );
       } else {
+        console.log('canSeeUsersInLeaderboard')
```
Comment (nathanah) [inline src/helpers/dashboardhelper.js:218]:
```suggestion
```
whoops

**Review by jennZhang93 — APPROVED** [review]
Working great! I've tested the functionality with both Volunteer and Core Team roles. I was able to Add/Delete the permission and see changes in the dashboard accordingly. Good job! [Frontend review](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2116#pullrequestreview-1984708586)

Volunteer:
Screenshot 2024-04-06 at 5 38 26 PM
Screenshot 2024-04-06 at 5 38 48 PM
Screenshot 2024-04-06 at 5 37 28 PM

Core Team:
Screenshot 2024-04-06 at 5 38 26 PM
Screenshot 2024-04-06 at 5 37 28 PM

**Comment by MQChen211** [comment]
Hi, I tested your PR with the frontend, and it works as expected.
[https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2116#issuecomment-2050359301](url)

**Pull request merged**